OpenRA.Mods.Common/Widgets/Logic/Ingame/AddFactionSuffixLogic.cs

```diff
@@ -45,6 +45,7 @@ public AddFactionSuffixLogic(Widget widget, World world)
 			{
 				ptw.ArrowButton += suffix;
 				ptw.TabButton += suffix;
+				ptw.Decorations += suffix;
```

Comment:
This has no effect because the `Initialize` method has already run and baked the non-suffixed `Decorations` into the cached transform. Not sure how to go about solving this - maybe we need to change the cached transform returned from `WidgetUtils.GetCachedStatefulImage` to also track the image name.

Reply:
Yes, you're right, I overlooked that one. So what do you suggest? Should I create new PR with change to `GetCachedStatefulImage`, where I add image name to the `CachedTransform`'s input? Or is there anything that might be negatively affected by this change?

Reply:
> maybe we need to change the cached transform returned from `WidgetUtils.GetCachedStatefulImage` to also track the image name.

A better solution might be a RefreshCaches function in `ProductionTabsWidget`. Or perhaps a reimplementation of `AddFactionSuffixLogic` to update caches instead of string values.

For this PR I don't think we need to solve this

Reply:
Should I remove the line from `AddFactionSuffixLogic` then? It does not matter, I think.

Reply:
IMO you should leave it. It needs to be fixed rather than removed. It's better to add a comment

```C#
// TODO: This isn't functional, ProductionTabsWidget's caches aren't updated with the new values.
```

Reply:
Yeah, we don't need to solve this here but we should open an issue to track it. It also affects `ScrollPanelWidget` in the same way.